### PR TITLE
Improving Decompiler Widget - GSoC project blog

### PR DESCRIPTION
**Pull request by NirmalManoj**
**Detailed description**
Important things that are left are: 
1. An image for the post
2. My image thumbnail.

...

**Test plan**
1. Read and tell me if I  should improve it, and in what ways.
2. Make sure I haven't missed anything that's must be there in this.
3. Check if I made all the changes required for a new blog post.

...

**Closing issues**

...

### CODE REVIEW

File: _posts/2020-08-31-improving-decompiler-widget-gsoc.md [hunk 1]
```diff
@@ -84,30 +85,36 @@ GIF demonstrating rename function action:
 3. cutter [PR #2286](https://github.com/radareorg/r2ghidra-dec/pull/123): Action to rename functions in the decompiler context menu
 
 ## Implementing annotations for global variables, constant variables, and related actions 
-This involved consolidating all references (function names, global variables, and constant variables) under a single structure called the reference. Two new types were also introduced to accommodate global variables and constant variables. Also, annotators were implemented for these new types of annotations in r2ghidra-dec. 
+This involved consolidating all references (function names, global variables, and constant variables) under a single structure called a `reference`. Two new types were also introduced to accommodate global variables and constant variables. Also, annotators were implemented for these new types of annotations in r2ghidra-dec.
```
Comment (ITAYC0HEN) [inline _posts/2020-08-31-improving-decompiler-widget-gsoc.md:88]:
```suggestion
This involved consolidating all references (function names, global variables, and constant variables) under a single structure called a "reference". Two new annotation types were also introduced to accommodate global variables and constant variables. Also, annotators were implemented for these new types of annotations in r2ghidra-dec. 
```

**Review by ITAYC0HEN — CHANGES_REQUESTED** [review]
Very nice! :)
Left a second round of comments

File: _posts/2020-08-31-improving-decompiler-widget-gsoc.md (hunk 1 above)
Comment (NirmalManoj) [reply _posts/2020-08-31-improving-decompiler-widget-gsoc.md]: Changed as suggested.

File: _posts/2020-08-31-improving-decompiler-widget-gsoc.md [hunk 2]
```diff
@@ -44,36 +44,37 @@ This involved refactoring the decompiler widget to use `RAnnotatedCode` instead
...
+1. cutter [PR #2256](/assets/images/blog/posts/improving-decompiler-widget-gsoc/skeleton-copy.gif): Dedicated decompiler context menu (skeleton) with an action to copy selected code
 
 ## Implementing actions for debugging 
-This involved me implementing actions for toggling breakpoint, accessing menu for advanced breakpoint, and also for two other debugging related actions in the decompiler context menu. Multiple breakpoints could exist in a line of decompiled code, as that’s often a more straightforward representation of multiple lines in assembly. The option to choose the breakpoint that a user wants to edit, from the list of breakpoints in the line was also implemented. 
+This involved me implementing actions for the toggling breakpoint, accessing the menu for advanced breakpoint definition, and also for two other debugging related actions in the decompiler context menu. Multiple breakpoints could exist in a decompiled code line, as that’s often a more straightforward representation of multiple lines in assembly. The option to choose the breakpoint that a user wants to edit from the breakpoints list was also implemented.
```
[the remark this answers is not in the document]
Comment (NirmalManoj) [reply _posts/2020-08-31-improving-decompiler-widget-gsoc.md]: Changed as suggested.

File: _posts/2020-08-31-improving-decompiler-widget-gsoc.md [hunk 3]
```diff
@@ -44,36 +44,37 @@ This involved refactoring the decompiler widget to use `RAnnotatedCode` instead
 This involves me making a context-menu with a single action for the decompiler. I replaced the disassembly context menu completely in the decompiler widget. The only action implemented along with the context menu skeleton was an action to copy selected decompiled code.
 
 GIF that shows the decompiler context menu with just copy action
-![skeleton-copy](/assets/images/blog/posts/improving-decompiler-widget-gsoc/skeleton-copy.gif)
+![skeleton-copy](https://user-images.githubusercontent.com/18501167/91670140-51207900-eb38-11ea-86dd-89101a9dbe89.gif)
 
 ### Related PR:
-1. cutter [PR #2256](https://github.com/radareorg/cutter/pull/2256): Dedicated decompiler context menu (skeleton) with an action to copy selected code
+1. cutter [PR #2256](/assets/images/blog/posts/improving-decompiler-widget-gsoc/skeleton-copy.gif): Dedicated decompiler context menu (skeleton) with an action to copy selected code
```
[the remark this answers is not in the document]
Comment (NirmalManoj) [reply _posts/2020-08-31-improving-decompiler-widget-gsoc.md]: This was an unintended change, fixed.

File: _posts/2020-08-31-improving-decompiler-widget-gsoc.md [hunk 4]
```diff
@@ -44,36 +44,37 @@ This involved refactoring the decompiler widget to use `RAnnotatedCode` instead
 This involves me making a context-menu with a single action for the decompiler. I replaced the disassembly context menu completely in the decompiler widget. The only action implemented along with the context menu skeleton was an action to copy selected decompiled code.
 
 GIF that shows the decompiler context menu with just copy action
-![skeleton-copy](/assets/images/blog/posts/improving-decompiler-widget-gsoc/skeleton-copy.gif)
+![skeleton-copy](https://user-images.githubusercontent.com/18501167/91670140-51207900-eb38-11ea-86dd-89101a9dbe89.gif)
```
[the remark this answers is not in the document]
Comment (NirmalManoj) [reply _posts/2020-08-31-improving-decompiler-widget-gsoc.md]: This was an unintended change, fixed.

File: _posts/2020-08-31-improving-decompiler-widget-gsoc.md [hunk 5]
```diff
@@ -10,27 +10,27 @@ description: "The decompiler widget is now responsive than ever! Read about the
...
 
-Cutter supports three decompilers: R2Ghidra(plugin for Ghidra), R2Dec, and R2RetDec(plugin for RetDec). The output from all these decompilers was being parsed into a custom structure named `AnnotatedCode`. The cutter’s decompiler used this. Even then, internally, r2ghidra-dec and r2retdec used `RAnnotatedCode`. After I shifted `RAnnotatedCode` to radare2, I sent patches to both r2ghidra-dec and r2retdec to use `RAnnotatedCode` directly from radare2.
+Cutter supports three decompilers: [R2Ghidra](https://github.com/radareorg/r2ghidra-dec)(plugin for Ghidra), [R2Dec](https://github.com/radareorg/r2dec-js), and [RetDec-Radare2](https://github.com/avast/retdec-r2plugin)(plugin for RetDec). The output from all these decompilers was being parsed into a custom structure name `AnnotatedCode` to use in Cutter.  Even then, internally, r2ghidra-dec and retdec-radare2 used `RAnnotatedCode`. After I shifted `RAnnotatedCode` to radare2, I sent patches to both r2ghidra-dec and retdec-radare2 to use `RAnnotatedCode` directly from radare2.
```
[the remark this answers is not in the document]
Comment (NirmalManoj) [reply _posts/2020-08-31-improving-decompiler-widget-gsoc.md]: Changed as suggested

File: _posts/2020-08-31-improving-decompiler-widget-gsoc.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
+---
+layout: post
+title: "Improving Decompiler Widget - GSoC Project"
+date: 2020-05-20
+tags: ["Decompiler", "GSoC"]
+categories: ["Cutter", "Decompiler", "GSoC"]
+author: nirmal_manoj
+# post_image: "/assets/images/blog/posts/improving-decompiler-widget-gsoc/decompiler_1.png"
```
Comment (ITAYC0HEN) [inline _posts/2020-08-31-improving-decompiler-widget-gsoc.md:8]: what about this?

File: _data/authors.yml [hunk 7]
```diff
@@ -12,4 +12,9 @@ cutter:
...
     website: https://cutter.re
     bio: The team behind Cutter.
-    image: /assets/images/blog/authors/cutter.png
\ No newline at end of file
+    image: /assets/images/blog/authors/cutter.png
+
+nirmal_manoj:
+    name: Nirmal Manoj
+    bio: Nirmal Manoj is a CS undergraduate student at IIIT Hyderabad, India. He was a GSoC 2020 student who worked on improving the decompiler widget.
+    # image: /assets/images/blog/authors/nirmal_manoj.png
```
Comment (ITAYC0HEN) [inline _data/authors.yml:20]: what about this?

File: _posts/2020-08-31-improving-decompiler-widget-gsoc.md (hunk 6 above)
Comment (NirmalManoj) [reply _posts/2020-08-31-improving-decompiler-widget-gsoc.md:8]: @ITAYC0HEN Please suggest some choices for the post image.

File: _data/authors.yml (hunk 7 above)
Comment (NirmalManoj) [reply _data/authors.yml:20]: I will upload it.

File: _posts/2020-08-31-improving-decompiler-widget-gsoc.md (hunk 6 above)
Comment (ITAYC0HEN) [reply _posts/2020-08-31-improving-decompiler-widget-gsoc.md:8]: decompiled code :)

**Review by ITAYC0HEN — APPROVED** [review]

**Pull request merged**